### PR TITLE
[pkg/traslator] move skywalking_to_traces into pkg/translator

### PR DESCRIPTION
**Pull request by JaredTan95**
**Description:** A part of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28693

move`skywalking_to_traces` in `skywalkingreceiver` into `pkg/translator/skywalking`

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper_windows.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func stopSampling(_ context.Context) error {
 		// no load scraper is running nothing to do
 		return nil
 	}
-	
+
```
Comment (fatsheep9146) [inline receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper_windows.go:143]: this change seems unrelated

Comment (JaredTan95) [reply receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper_windows.go:143]: I did not chang it manually, seems changed by `make genotelcontribcol  `

Comment (JaredTan95) [reply receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper_windows.go:143]: Do I need  revert this line ?

Comment (fatsheep9146) [reply receiver/hostmetricsreceiver/internal/scraper/loadscraper/load_scraper_windows.go:143]: No, I think this is not a big deal

**Review by fatsheep9146 — APPROVED** [review]

File: pkg/translator/skywalking/go.mod [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/skywalking
+
+go 1.20
+
+require (
+	github.com/google/uuid v1.4.0
+	github.com/stretchr/testify v1.8.4
+	go.opentelemetry.io/collector/pdata v1.0.0-rcv0017.0.20231026220224-6405e152a2d9
+	go.opentelemetry.io/collector/semconv v0.88.1-0.20231026220224-6405e152a2d9
+	skywalking.apache.org/repo/goapi v0.0.0-20231026090926-09378dd56587
```
Comment (mx-psi) [inline pkg/translator/skywalking/go.mod:10]: This changed from ` v0.0.0-20220121092418-9c455d0dda3f` to `v0.0.0-20231026090926-09378dd56587`. Was this intentional?

Comment (JaredTan95) [reply pkg/translator/skywalking/go.mod:10]: https://github.com/apache/skywalking-goapi/releases
Yes, There is no official version yet.

Comment (mx-psi) [reply pkg/translator/skywalking/go.mod:10]: Using a pseudoversion is fine, I just meant we are jumping from something from January 2022 up to something from October 2023, wondering if the diff between those versions is fine

Comment (JaredTan95) [reply pkg/translator/skywalking/go.mod:10]: Changes from January 2022 to October 2023 We didn't used, and we did UT to cover what we used, which was fine to me.

**Review by mx-psi — APPROVED** [review]
